.github/workflows/build-and-test.yml

```
@@ -0,0 +1,51 @@
+name: facebook/metro/build-and-test
+on:
+  pull_request:
+    types: [opened, synchronize]
+
+# head_ref is per PR, so this ensures that updating the latest PR commit
+# will cancel the previous run of the workflow and trigger a new one
+concurrency:
+  group: "build-and-test-${{ github.head_ref }}"
+  cancel-in-progress: true
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  run-js-checks:
+    runs-on: ubuntu-latest
+    name: "Typescript, Lint, Smoke Test"
+    steps:
+      - uses: actions/checkout@v4
+      - uses: ./.github/actions/yarn-install
+      - run: yarn typecheck
+      - run: yarn typecheck-ts
+      - run: yarn lint
+      - run: yarn test-smoke
+
+  test-with-coverage:
+    runs-on: ubuntu-latest
+    name: "Test Coverage Calculation"
+    steps:
+    - uses: actions/checkout@v4
+    - uses: ./.github/actions/yarn-install
+    - run: yarn test-coverage
+    - run: "./.github/scripts/install_codecov.sh"
+    - run: "./codecov -t ${{ secrets.CODECOV_TOKEN }} -f ./coverage/coverage-final.json"
+
+  test:
+    strategy:
+      matrix:
+        runs-on: ['ubuntu-latest'] # 'windows-latest'
+        node-version-type: ['current', 'min-supported']
+    name: "Tests [${{ matrix.runs-on }} machine, ${{ matrix.node-version-type }} Node.js]"
```

nit, also ideally let's print the numeric Node version, e.g. `Tests [Node.js 20, ubuntu-latest]`

```suggestion
    name: "Tests [Node.js ${{ matrix.node-version-type }}, ${{ matrix.runs-on }}]"
```

---

I'm in favour of eliminating `node-version-type` and just using a `node-version` matrix, which still supports aliases like `lts/*`, `lts/-n`, `current`.